node/src/actors/chain_manager/mod.rs

```diff
@@ -206,6 +206,10 @@ pub struct ChainManager {
     temp_superblock_votes: Vec<SuperBlockVote>,
     /// Commits and reveals to process later
     temp_commits_and_reveals: Vec<Transaction>,
+    /// Value transfers and data requests to process later
+    temp_vts_and_drs: Vec<Transaction>,
```

Comment:
Since we will be writing transactions to the back of this vector and reading them from the front, like a FIFO queue, the logical choice is to use a `VecDeque`.

```suggestion
    temp_vts_and_drs: VecDeque<Transaction>,
```

The main benefit is that `drain` will not need to move the remaining elements. Example with 4 elements, A, B, C, D, and removing the first 2:

```
Using Vec:
ABCD
.drain(..2)
CD
^ performs a memcpy to move CD to the position of AB

Using VecDeque
ABCD
.drain(..2)
  CD
^ does not copy anything, CD is still at the same position in memory
```